src/backend/distributed/executor/multi_utility.c

```diff
@@ -391,6 +399,27 @@ multi_ProcessUtility(Node *parsetree,
 	{
 		ListCell *ddlJobCell = NULL;
 
+		/*
+		 * Check the altered table's constraints using the same logic with create table.
+		 * Note that we already eliminated alter table command with more than one
+		 * sub-command with ErrorIfUnsupportedAlterTableStmt function. Therefore,
+		 * controlling the type of first(and single) sub-command of command list
+		 * is enough.
+		 */
+		if (IsA(parsetree, AlterTableStmt))
+		{
+			AlterTableStmt *alterTableStatement = (AlterTableStmt *) parsetree;
```

Comment:
Let's add here an Assert to ensure commandList really has single sub-command, similar to [here](https://github.com/citusdata/citus/blob/master/src/backend/distributed/executor/multi_utility.c#L944)